### PR TITLE
P0 components

### PR DESCRIPTION
**Pull request by ritu9898**

### CODE REVIEW

File: components/avatar.html [hunk 1]
```diff
@@ -18,16 +18,16 @@ <h2>Default</h2>
...
-        <img src="../components/images/default-avatar2.png" class="default-avatar" />
-        <!-- <svg width="100" height="100">
-          <circle cx="50" cy="50" r="40" stroke="grey" stroke-width="4" fill="white" />
-        </svg> -->
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div>
+            <img src="../components/images/default-avatar2.png" class="default-avatar" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/avatar.html [hunk 2]
```diff
@@ -57,15 +61,18 @@ <h2>Appearance(Sqaure)</h2>
...
-        <img src="../components/images/Ritu.jpeg" class="square-avatar" />
-        <img src="../components/images/square-avatar.png" class="square-avatar" />
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div class="appearance-avatar">
+            <img src="../components/images/default-avatar2.png" class="square-avatar" />
+            <img src="../components/images/Ritu.jpeg" class="square-avatar" />
+            <img src="../components/images/square-avatar.png" class="square-avatar" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/avatar.html [hunk 3]
```diff
@@ -79,16 +86,19 @@ <h3>xxlarge</h3>
...
-          <!-- <img src="../components/images/Ritu.jpeg" class="xxlarge" /> -->
-          <img src="../components/images/square-avatar.png" class="xxlarge" />
-        </div>
-        <div>
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="appearance-avatar">
+              <img src="../components/images/default-avatar2.png" class="square-avatar xxlarge" />
+              <img src="../components/images/Ritu.jpeg" class="circle-avatar xxlarge" />
+              <img src="../components/images/square-avatar.png" class="xxlarge" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/avatar.html [hunk 4]
```diff
@@ -99,16 +109,19 @@ <h3>xlarge</h3>
...
-          <!-- <img src="../components/images/Ritu.jpeg" class="xlarge" /> -->
-          <img src="../components/images/square-avatar.png" class="xlarge" />
-        </div>
-        <div>
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="appearance-avatar">
+              <img src="../components/images/default-avatar2.png" class="square-avatar xlarge" />
+              <img src="../components/images/Ritu.jpeg" class="circle-avatar xlarge" />
+              <img src="../components/images/square-avatar.png" class="xlarge" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/avatar.html [hunk 5]
```diff
@@ -119,16 +132,19 @@ <h3>large</h3>
...
-          <!-- <img src="../components/images/Ritu.jpeg" class="large" /> -->
-          <img src="../components/images/square-avatar.png" class="large" />
-        </div>
-        <div>
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="appearance-avatar">
+              <img src="../components/images/default-avatar2.png" class="square-avatar large" />
+              <img src="../components/images/Ritu.jpeg" class="circle-avatar large" />
+              <img src="../components/images/square-avatar.png" class="large" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/avatar.html [hunk 6]
```diff
@@ -139,22 +155,23 @@ <h3>medium</h3>
...
-          <!-- <img src="../components/images/Ritu.jpeg" class="medium" /> -->
-          <img src="../components/images/square-avatar.png" class="medium" />
-        </div>
-        <div>
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="appearance-avatar">
+              <img src="../components/images/default-avatar2.png" class="square-avatar medium" />
+              <img src="../components/images/Ritu.jpeg" class="circle-avatar medium" />
+              <img src="../components/images/square-avatar.png" class="medium" />
```
Comment (zoef-yt) [inline components/avatar.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/badge.css [hunk 7]
```diff
@@ -0,0 +1,52 @@
+.default-badge, .badge-on-avatar {
+  height: 1.5rem;
+  width: 1.5rem;
+  border-radius: 50%;
+  background-color: silver;
```
Comment (zoef-yt) [inline components/badge.css]: Change:- Use var for colour.

File: components/badge.css [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+.cart-badge {
+  height: 3.5rem;
+  width: 3.5rem;
+}
+
+.badge-on-avatar {
+  position: absolute;
+  background-color: chartreuse;
```
Comment (zoef-yt) [inline components/badge.css]: Change:- Use var for colour.

File: components/badge.css [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  right: 0;
+  bottom: 0;
+  left: 30px;
+  top: -6px;
+}
+
+.notification-badge {
+  background-color: red;
+  color: whitesmoke;
```
Comment (zoef-yt) [inline components/badge.css]: Change:- Use var for colour.

File: components/badge.css [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  bottom: 0;
+  left: 40px;
+  top: -9px;
+}
+
+.badge-on-card {
+  width: auto;
+  background-color: bisque;
+  padding: 0.5rem 0.5rem;
+  /* margin: 2rem 0px; */
```
Comment (zoef-yt) [inline components/badge.css]: Change:- Remove commented code.

File: components/badge.css [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  background-color: red;
+  color: whitesmoke;
+  font-size: 10px;
+}
+
+.cart-on-badge {
+  position: absolute;
+  right: 0;
+  bottom: 0;
+  left: 40px;
```
Comment (zoef-yt) [inline components/badge.css]: Change:-  Use relative units.

File: components/badge.html [hunk 12]
```diff
@@ -36,14 +37,18 @@ <h2>Avatar badge for status</h2>
...
       </p>
-      <div class="avatar-badge">
-        <img src="../components/images/Ritu.jpeg" class="circle-avatar avatar-badge" />
-        <section class="badge-on-avatar"></section>
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div class="avatar-badge">
+            <img src="../components/images/Ritu.jpeg" class="circle-avatar avatar-badge" />
```
Comment (zoef-yt) [inline components/badge.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/badge.html [hunk 13]
```diff
@@ -54,14 +59,18 @@ <h2>Avatar badge for notification</h2>
...
       </p>
-      <div class="avatar-badge">
-        <img src="../components/images/Ritu.jpeg" class="circle-avatar avatar-badge" />
-        <section class="badge-on-avatar notification-badge">20+</section>
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div class="avatar-badge">
+            <img src="../components/images/Ritu.jpeg" class="circle-avatar avatar-badge" />
```
Comment (zoef-yt) [inline components/badge.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/badge.html [hunk 14]
```diff
@@ -72,18 +81,22 @@ <h2>Avatar badge for notification</h2>
...
       </p>
-      <div class="avatar-badge">
-        <img src="../components/images/cart.png" class="cart-badge" />
-        <section class="badge-on-avatar cart-on-badge">5</section>
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div class="avatar-badge">
+            <img src="../components/images/cart.png" class="cart-badge" />
```
Comment (zoef-yt) [inline components/badge.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/card.html [hunk 15]
```diff
@@ -5,9 +5,6 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <title></title>
```
Comment (zoef-yt) [inline components/card.html:7]: The title tag is empty.

File: components/card.html [hunk 16]
```diff
@@ -21,20 +18,29 @@ <h2>Default</h2>
...
-          </div>
-          <div>
-            <p class="title">Avatar</p>
-            <p>An avatar is a visual representation of a user or entity.</p>
+
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="card">
+              <div class="child-card">
+                <img src="../components/images/default-avatar2.png" class="default-avatar" />
```
Comment (zoef-yt) [inline components/card.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/card.html [hunk 17]
```diff
@@ -45,23 +51,35 @@ <h2>Card with Badge</h2>
...
-          <div class="child-card">
-            <img src="../components/images/chair.png" class="" />
-            <div class="badge-on-card">
-              #1 Bestseller
+
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="card">
+              <div class="child-card">
+                <img src="../components/images/chair.png" class="" />
```
Comment (zoef-yt) [inline components/card.html]: Suggestion:- This is no alt attribute please add alt attribute. Class attribute is also empty.

File: components/card.html [hunk 18]
```diff
@@ -72,26 +90,41 @@ <h2>Card with Text overlay</h2>
...
-              <img src="../components/images/chair.png" class="" />
-              <div class="badge-on-card">
-                #1 Bestseller
+
+        <div class="code-border">
+          <div class="code-bg">
+            <div class="card">
+              <div class="child-card">
+                <div class="transparent">
+                  <img src="../components/images/chair.png" class="" />
```
Comment (zoef-yt) [inline components/card.html]: Suggestion:- This is no alt attribute please add alt attribute. The class attribute is also empty.

File: components/image.html [hunk 19]
```diff
@@ -9,23 +9,25 @@
...
       </p>
-      
-      <div class="image">
-        <img src="../assets/images/moon.jpg" />
-      </div>
-      <div>
+      <div class="code-border">
+        <div class="code-bg">
+          <div class="">
+            <img src="../assets/images/flower1.jpg" class="img-fluid" />
```
Comment (zoef-yt) [inline components/image.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/image.html [hunk 20]
```diff
@@ -9,23 +9,25 @@
...
+          </div>
+        </div>
         <code>
-          <span><</span>div class="default-avatar"<span>></span>
-          <span><</span>div<span>></span>
+          <xmp>
+            <img src="../assets/images/flower1.jpg" class="img-fluid" />
```
Comment (zoef-yt) [inline components/image.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/image.html [hunk 21]
```diff
@@ -34,16 +36,17 @@ <h2>Default</h2>
...
       </p>
-      
-      <div class="image">
-        <img src="../assets/images/moon.jpg" class="circle" />
-      </div>
-      <div>
+
+      <div class="code-border">
+        <div class="code-bg">
+            <img src="../assets/images/flower.jpg" class="img-fluid circle" />
```
Comment (zoef-yt) [inline components/image.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/image.html [hunk 22]
```diff
@@ -52,16 +55,17 @@ <h2>Circle Image</h2>
...
       </p>
-      
-      <div class="image object">
-        <img src="../assets/images/moon.jpg" class="circle" />
-      </div>
-      <div>
+
+      <div class="code-border">
+        <div class="code-bg">
+          <img src="../assets/images/moon.jpg" class="circle object" />
```
Comment (zoef-yt) [inline components/image.html]: Suggestion:- This is no alt attribute please add alt attribute.

File: components/style/alert.css [hunk 23]
```diff
@@ -18,28 +18,6 @@
...
-}
-
-.code-border {
-  border-radius: 4px;
-  box-shadow: 0 0 0 1.2px var(--ds-border-neutral,#ebecf0);
-}
-
 .info {
   color: #31708f;
   background-color: #d9edf7;
```
Comment (zoef-yt) [inline components/style/alert.css]: Change:- Use var for colour.

File: components/style/input.css [hunk 24]
```diff
@@ -0,0 +1,55 @@
+.input-group input{
+  display: block;
+  padding: 0.375rem 0.75rem;
+  font-size: 1rem;
+  line-height: 1.5;
+  color: #495057;
+  background-color: #fff;
+  border: 1px solid #ced4da;
```
Comment (zoef-yt) [inline components/style/input.css]: Change:- Use var for colour.

File: components/style/input.css [hunk 25]
```diff
@@ -0,0 +1,55 @@
...
+  transition: border-color .15s ease-in-out,box-shadow .15s ease-in-out;
+  margin: 1rem 0;
+}
+
+.input-group input:focus {
+  color: #495057;
+  background-color: #fff;
+  border-color: #80bdff;
+  outline: 0;
+  box-shadow: 0 0 0 0.2rem rgb(0 123 255 / 25%);
```
Comment (zoef-yt) [inline components/style/input.css]: Change:- Use var for colour and stay consistent with the colour.

File: components/style/input.css [hunk 26]
```diff
@@ -0,0 +1,55 @@
...
+  opacity: 1;
+}
+
+.invalid-input {
+  border-color: #dc3545 !important;
+}
+
+.input-group .invalid-input:focus {
+  border-color: #dc3545;
+  box-shadow: 0 0 0 0.2rem rgb(220 53 69 / 25%);
```
Comment (zoef-yt) [inline components/style/input.css]: Change:- Use var for colour.

File: components/style/input.css [hunk 27]
```diff
@@ -0,0 +1,55 @@
...
+
+.input-group .input-sm {
+  padding: 0.25rem 0.5rem;
+  font-size: .875rem;
+  line-height: 1.5;
+  border-radius: 0.2rem;
+}
+
+.input-group .input-disabled {
+  background-color: #e9ecef;
```
Comment (zoef-yt) [inline components/style/input.css]: Change:- Use var for colour.

File: index.css [hunk 28]
```diff
@@ -104,4 +103,24 @@ code {
...
+  135deg
+  ,transparent 75%,var(--ds-background-sunken,#f9f9fa) 75% );
+      background-size: 20px 20px;
+      background-position: 0px 0px, 10px 0px, 10px -10px, 0px 10px;
+      border-radius: 3px 3px 0px 0px;
+}
+
+.code-border {
+  border-radius: 4px;
+  box-shadow: 0 0 0 1.2px var(--ds-border-neutral,#ebecf0);
```
Comment (zoef-yt) [inline index.css:173]: Change:- Use var for colour.

File: index.css [hunk 29]
```diff
@@ -1,20 +1,19 @@
...
 body {
   font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
 }
 
 #header .nav {
   display: flex;
   padding: 2rem;
-  /* background-color: rgb(9, 30, 66); */
   background-image: linear-gradient(135deg, #1da1f2, #981ceb);
   background-color: #1b95e0;
```
Comment (zoef-yt) [inline index.css]: Change:- Use var for colour.

File: components/style/input.css [hunk 30]
```diff
@@ -0,0 +1,55 @@
...
+.input-group .invalid-input:focus {
+  border-color: #dc3545;
+  box-shadow: 0 0 0 0.2rem rgb(220 53 69 / 25%);
+}
+
+.invalid-input-msg {
+  width: 100%;
+  margin-top: 0.25rem;
+  font-size: 80%;
+  color: #dc3545;
```
Comment (zoef-yt) [inline components/style/input.css]: Change:- Use var for colour.

**Review by zoef-yt — CHANGES_REQUESTED** [review]
Many places need changes. Also, this branch has conflicts so make sure to resolve them too.

File: components/style/input.css (hunk 26 above)
Comment (ritu9898) [reply components/style/input.css]: I'd used different colors in different branches, thus will convert them into variables in the final merge.

**Review by sanu29 — APPROVED** [review]
Good to merge for now

**Pull request closed without merging**